Add secrets reference (#2125)
sensu/sensu-docs: content/sensu-go/5.17/reference/secrets.md

 
## Secret specification

### Top-level attributes

type         | 
-------------|------
description  | Top-level attribute that specifies the resource type. For secrets configuration, the type should always be `Secret`.
required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
type         | String
example      | {{< highlight shell >}}"type": "Secret"{{< /highlight >}}

api_version  | 
-------------|------
description  | Top-level attribute that specifies the Sensu API group and version. For secrets configuration in this version of Sensu, the `api_version` should always be `secrets/v1`.
required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
type         | String
example      | {{< highlight shell >}}"api_version": "secrets/v1"{{< /highlight >}}

metadata     |      |
-------------|------
description  | Top-level scope that contains the secret's `name` and `namespace`.
required     | true
type         | Map of key-value pairs
example      | {{< highlight shell >}}
"metadata": {

>   "name": "sensu-ansible-token",

Author: ```suggestion
  "name": "sensu-ansible",
```

  "namespace": "default"
}
{{< /highlight >}}

spec         | 
-------------|------
description  | Top-level map that includes secrets configuration [spec attributes][8].
required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
type         | Map of key-value pairs
example      | {{< highlight shell >}}
"spec": {

>   "id": "ANSIBLE_TOKEN",

Author: ```suggestion
  "id": "secret/ansible#token",
```

example      | {{< highlight shell >}}"id": "secret/ansible#token"{{< /highlight >}}

provider     | 
-------------|------ 
description  | The name of the Sensu provider with the secret.
required     | true
type         | String
example      | {{< highlight shell >}}"provider": "ansible_vault"{{< /highlight >}}

## Secret configuration

Use the [Secrets API][2] and [sensuctl][3] to create, view, and manage your secrets configuration.
To manage secrets configuration with sensuctl, configure sensuctl as the default [`admin` user][6].

The [standard sensuctl subcommands][4] are available for secrets (list, info, and delete).

To list all secrets:

{{< highlight shell >}}
sensuctl secret list
{{< /highlight >}}

To see a secret's status:

{{< highlight shell >}}
sensuctl secret info SECRET_NAME
{{< /highlight >}}

To delete a secret:

{{< highlight shell >}}
sensuctl secret delete SECRET_NAME
{{< /highlight >}}

## Secret examples

A secret resource definition refers to a secrets `id` and a secrets `provider`.
Read the [secrets provider reference][7] for the provider specification.

{{< language-toggle >}}

{{< highlight yml >}}
---
type: Secret
api_version: secrets/v1
metadata:

>   name: sensu-ansible-token

Author: ```suggestion
  name: sensu-ansible
```

  namespace: default
spec:

>   id: ANSIBLE_TOKEN

Author: ```suggestion
  id: 'secret/ansible#token'
```

  provider: env
{{< /highlight >}}

{{< highlight json >}}
{
  "type": "Secret",
  "api_version": "secrets/v1",
  "metadata": {

>     "name": "sensu-ansible-token",

Author: ```suggestion
    "name": "sensu-ansible",
```

    "namespace": "default"
  },
  "spec": {
    "id": "ANSIBLE_TOKEN",
    "provider": "env"
  }
}
{{< /highlight >}}

{{< /language-toggle >}}

Secrets that target a HashiCorp Vault always have an `id` as shown in the following example:

{{< language-toggle >}}

{{< highlight yml >}}
---
type: Secret
api_version: secrets/v1
metadata:
  name: sensu-ansible-token
  namespace: default
spec:
  id: 'secret/website#database'
  provider: ansible_vault
{{< /highlight >}}

{{< highlight json >}}
{
  "type": "Secret",
  "api_version": "secrets/v1",
  "metadata": {
    "name": "database",
    "namespace": "default"
  },
  "spec": {
    "id": "secret/website#database",
    "provider": "vault"
  }
}
{{< /highlight >}}

{{< /language-toggle >}}

> Secrets that target a HashiCorp Vault must start with the word `secret`.

Author: ```suggestion
The `id` value for secrets that target a HashiCorp Vault must start with the name of the secret's path in Vault.
This example uses the `secret` path.
```

Author: Is this suggestion correct? It seems like all the "secrets" in this explanation could be confusing.
